### ISSUE TITLE
FNX3-16352 ⁃ Force enable zoom

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#2477]
## User Story
* As a user with low vision, I want to be able to override a website's request to prevent zooming in, so I can always read the content on a page. 

## Dependencies
* Needs GV API: https://bugzilla.mozilla.org/show_bug.cgi?id=1540615

### Acceptance Criteria
-I can zoom in on every page
-I can enable the "Always enable zoom" setting in my browser (for accessibility settings, because some sites can be problematic)

**Comment by Lululalu** [comment]
It looks the same as #1372

**Comment by vesta0** [comment]
Thank you @Lululalu ! I had totally forgotten that I had filed this before!

**Comment by sblatz** [comment]
Let's make sure this is properly prioritized with GV

@bifleming, @cpeterso 

UX to decide how this toggle will look in the accessibility settings.

**Comment by cpeterso** [comment]
> Let's make sure this is properly prioritized with GV

The GV zoom bug is a P2, meaning we will try to work on it (some time) in Q4. If you would like this zoom API delivered in early Q4 (like October), please let me know.

**Comment by liuche** [comment]
@cpeterso if you could get ready for us earlier that would be really helpful!

**Comment by cpeterso** [comment]
> @cpeterso if you could get ready for us earlier that would be really helpful!

OK. I increased the GV bug's priority rank so we will consider it for GV's October sprint.

**Comment by cpeterso** [comment]
The GV bug fix will be in tomorrow's GV 71 Nightly, so I will remove this issue's `needs:gv` and `waiting` labels.

https://bugzilla.mozilla.org/show_bug.cgi?id=1540615#c10

**Comment by liuche** [comment]
Adding this to the UX backlog since we wanted to do this earlier, and were waiting on this showing up in GV.

We need UX to decide how this toggle will look in the accessibility settings (because this could break some sites, so we don't want to turn it on by default).

**Comment by boek** [comment]
This feature is behind a feature flag enabled on Nightly

**Comment by softvision-miralobontiu** [comment]
@boek, this feature is not displayed in Settings - Accessibility so far.

I set the browser.ui.zoom.force-user-scalable pref from about:config to true, and it works. Pages like instagram.com, facebook.com can be zoomed after this pref was set to "true". When the pref is set to "false", those pages cannot be zoomed.

Tested with the Nightly build 10/23 from Play Store (#12960608) with Huawei P9 Lite (Android 7), Google Pixel (Android Q), and Samsung Galaxy Tab S3 (Android 8).

I will remove the qa-needed label, but not going to close this issue until we know if there should be a an option in Settings for it.

**Comment by serovar** [comment]
Setting the pref to true in about:config does work, but the pref get resetted when the browser is closed.

**Comment by AndiAJ** [comment]
Hi, I've just tested this issue on **the latest Nightly Build #12980627 from 25/10** and noticed the following:

1. If you close the app and afterwards navigate back to an Instagram or Facebook page the zoom doesn't seem to be working even if it's enabled in Accessibility.

Manage to reproduce this with a Google Pixel 3a XL (Android 9) Huawei and Mate 20 Lite (Android 8.0.1)
Not reproducible on Samsung Galaxy s7 (Android 7)

2. @serovar is right, it is impossible to zoom in from the area of an album 

@ekager @boek  - Please review the above. :smiley:

**Comment by neekless** [comment]
Can reproduce AndiAJ's problem on Nightly 191030 18:21 (Build #13031826).
Turning on Zoom on all websites in Accessibility will set `browser.ui.zoom.force-user-scalable` to true and zoom will work, but after force closing fenix `browser.ui.zoom.force-user-scalable` will reset to false and zoom will not work despite the setting still being on in Accessibility. The pref is not sticking.

**Comment by boek** [comment]
GV bug filed #6723

**Comment by boek** [comment]
Fixed bug where pref wasn't sticking. Passing back for QA :)

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest Nightly build from 11/27 with Samsung Galaxy Note 8 (Android 9), and HTC Desire 820 (Android 6.0.1). I've tested facebook and twitter pages. I was able to zoom in / out even on albums, on all areas. I've forced exit, and the pref remain active and worked correctly.

**Comment by rhjdvsgsgks** [comment]
`m.gamer.com.tw` still not scalable even enable "Zoom on all websites" and `browser.ui.zoom.force-user-scalable`
